Document how to use varargs in Argument Clinic (#1629)
python/devguide: development-tools/clinic.rst

a *type* argument to the object converter for :c:macro:`METH_O`.

How to convert var-positional parameter functions
-------------------------------------------------

> To convert a var-positional parameter function,
> prepending the parameter name with ``*`` ,
> and the parameter should use the ``object`` converter::

Reviewer: ```suggestion
To convert a var-positional parameter function, prepend the parameter name
with ``*`` and use the the ``object`` converter.
For example::
```

   /*[clinic input]
   var_positional_sample

       foo: int

>        *args: object

Author: ```suggestion
       *args: array
```

>        *args: object

Reviewer: Is this true? In `libclinic` I only see a `tuple` and `array` converter as subclasses of VarPosCConverter, and I can't see any uses of `*arg_name: object` in the tree, though there are plenty of uses of `*args : tuple` or `*args: array`.

a *type* argument to the object converter for :c:macro:`METH_O`.

> How to convert var-positional parameter functions
> -------------------------------------------------
> 
> To convert a var-positional parameter function, prepend the parameter name
> with ``*`` and use the the ``array`` converter.
> For example::
> 
>    /*[clinic input]
>    var_positional_sample
> 
>        foo: int
>        *args: array
>    [clinic start generated code]*/
> 
> The implementation function will receive var-positional arguments as C array
> *args* of :c:type:`PyObject * <PyObject>`.  Alternatively, you could use
> ``tuple`` converter to pass a regular :c:type:`PyTupleObject` as argument.

Reviewer: A suggested rephrasing to split `tuple` and `array`. I've also switched to saying `*args` instead of emphasising var-positional, the documentation uses `*args` or starargs, when discussing the feature in the language reference, it seems to be only internal to Clinic to say varpos.

```suggestion
How to convert ``*args`` parameters (starargs / var-positional)
---------------------------------------------------------------

There are two converters suitable for ``*args``: *array* and *tuple*.

Using the *array* converter will provide the implementation function with
a C array *args* of type of :c:type:`PyObject * <PyObject>` and the number
of items in the array as :c:type:`Py_ssize_t` *args_length*.
For example::

   /*[clinic input]
   var_positional_sample

       spam: int
       *args: array
   [clinic start generated code]*/

Using the *tuple* converter will provide the implementation function with
a standard :c:type:`PyTupleObject`.
For example::

   /*[clinic input]
   var_positional_sample

       spam: int
       *args: tuple
   [clinic start generated code]*/
```
